File: Python Course/Software Development Module - in Python/Revision/Decoder/DecoderApp.py
"""
DecoderApp.py
Billy Cussen
21/10/2020
"""

#Objects
import gc
import Decoder
from Decoder import myDecoder

#Variables
repeat = "y"
wordHistory = []

while repeat == "y":

    inputCheck = False

    #Input - Word?
    while inputCheck == False:
        try:
            word = str(input("Please enter the Word you would like to Decode: "))
            for c in word:
                if c == ' ':
                    raise Exception
            if len(word) <= 0:
                raise Exception
        except Exception as e:
            print("Please enter a valid word to be decoded (no spaces)")
        else:
            inputCheck = True

    #Process and Output
    myDecoderApp = myDecoder(word)
    print(myDecoderApp.decodeWord())
    wordHistory.append(myDecoderApp.decodeWord())
    myDecoderApp = None
    gc.collect()

    repeat = ""
    while repeat.lower() != "y" and repeat.lower() != "n":
        repeat = str(input("Please enter \"y\" to repeat and \"n\" to exit: "))
        if repeat.lower() != "y" and repeat.lower != "n":
            print("You need to enter \"y\" or \"n\"")


print("***WORD HISTORY***")
for word in wordHistory:
    print(word)

wordHistory=None
gc.collect()
print("\nThanks for using my Word Decoder App!")

File: Java/Software Development Module - Java/Revision/Decoder/DecoderApp.java
/*
DecoderApp.java
Billy Cussen
04/08/2020
*/

import java.util.Scanner;
import java.util.List;
import java.util.LinkedList;

public class DecoderApp{

    public static void main(String [] args){

        //Variables
        String inputWord = new String();
        String outputWord = new String();
        String repeatCheck;
        boolean inputCheck;
        
        //Object
        List<String> wordHistory = new LinkedList<String>();
        Scanner keyboard;
        Decoder myDecoder;

        //Input
        do{
        
            do{
                inputCheck = true;
                keyboard = new Scanner(System.in);

                System.out.print("Enter the Word to be Decoded: ");
                inputWord = keyboard.nextLine();

                for(int i = 0; i < inputWord.length(); i++){

                    if(inputWord.charAt(i) == ' '){
                        System.out.println("Please enter a valid Word!");
                        inputCheck = false;
                        keyboard = null;
                        break;
                    }

                }

            }while(!inputCheck);

            //Process
            myDecoder = new Decoder();
            outputWord = myDecoder.getDecodedWord(inputWord);
            wordHistory.add("Input Word "+(wordHistory.size()+1)+": "+inputWord+", Output Word: "+outputWord);
            keyboard = null;
            myDecoder = null;

            //Output
            System.out.println(outputWord+"\n");
            

            do{
                keyboard = new Scanner(System.in);


                System.out.print("Please enter if you would like to Continue(Y or N): ");
                repeatCheck = keyboard.nextLine();

                if(!(repeatCheck.equalsIgnoreCase("Y")||repeatCheck.equalsIgnoreCase("N"))){
                    System.out.println("Please enter Y or N!");
                    keyboard = null;
                }     

            }while(!(repeatCheck.equalsIgnoreCase("Y")||repeatCheck.equalsIgnoreCase("N")));

        }while(repeatCheck.equalsIgnoreCase("Y"));
        
        keyboard = null;

        System.out.println("\n***HISTORY***");
        for(String word: wordHistory){
            System.out.println(word);
        }

        wordHistory = null;

    }

}